core/src/main/java/io/aiven/klaw/helpers/db/rdbms/InsertDataJdbc.java

```diff
@@ -363,9 +376,13 @@ public String insertIntoClusters(KwClusters kwClusters) {
   public Integer getNextSeqIdAndUpdate(String entityName, int tenantId) {
     List<KwEntitySequence> kwEntitySequenceList =
         kwEntitySequenceRepo.findAllByEntityNameAndTenantId(entityName, tenantId);
-    Integer lastId = kwEntitySequenceList.get(0).getSeqId();
-    insertIntoKwEntitySequence(entityName, lastId + 1, tenantId);
-    return lastId;
+    if (!kwEntitySequenceList.isEmpty()) {
+      Integer lastId = kwEntitySequenceList.get(0).getSeqId();
+      insertIntoKwEntitySequence(entityName, lastId + 1, tenantId);
+      return lastId;
+    } else {
+      return null;
```

Comment:
if you return null here it gets incremented to 1 in line 661 EnvsClusterTenantsController. But that isn't updated in the DB.
would you be better off returning lastId++ and saving it to the db as well? 
That way you dont have to have the check in the calling method and it saved the initial value into the DB?
As you need to be able to cater for adding a new tenant aswell.